File: CocomeRegister/Models/Order.cs
using System.ComponentModel.DataAnnotations;

namespace CocomeStore.Models
{
    /// <summary>
    /// class <c>Order</c>
    /// </summary>
    public class Order : Trade<Provider>
    {
    }

    /// <summary>
    /// class <c>OrderElement</c>
    /// </summary>
    public class OrderElement : TradeElement
    {
        [Required]
        public int OrderId { get; set; }

        public Order Order { get; set; }
    }
}

File: CocomeRegister/ClientApp/src/models/Order.ts
import { Product } from "./Product";
import { Provider } from "./Provider";
import { Store } from "./Store";
import { Trade, TradeElement } from "./Trade";

export class OrderElement implements TradeElement {
    product: Product;
    amount: number;
}

export class Order implements Trade<Provider> {
    id: number;
    elements: TradeElement[];
    store: Store;
    provider: Provider;
    placingDate: Date;
    deliveringDate: Date;
    closed: boolean;
}
